server/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -249,7 +249,8 @@ protected AsyncShardFetch.FetchResult<NodeGatewayStartedShards> fetchData(ShardR
                     logger,
                     "shard_started",
                     shardId,
-                    IndexMetadata.INDEX_DATA_PATH_SETTING.get(allocation.metadata().index(shard.index()).getSettings())
+                    IndexMetadata.INDEX_DATA_PATH_SETTING.get(allocation.metadata().index(shard.index()).getSettings()),
+                    allocation.nodes().getSize()
```

Comment:
Just the data nodes:

```suggestion
                    allocation.routingNodes().size()
```